src/interfaces/simulation.ts

```diff
@@ -539,23 +535,24 @@ export class SimulationInterface<T extends ChainId> extends ServiceInterface<T>
 
     const zapProtocol = this.getZapProtocol({ vaultAddress: toVault });
 
-    const zapInParams = await this.yearn.services.zapper
-      .zapIn(from, zapToken, amount, toVault, options.gasPrice || "0", options.slippage, skipGasEstimate, zapProtocol)
+    const zapInParams = await this.yearn.services.portals
+      .zapIn(toVault, zapToken, amount, from, options.slippage, skipGasEstimate)
       .catch(() => {
-        throw new ZapperError("zap in", ZapperError.ZAP_IN);
+        throw new ZapError("zap in", ZapError.ZAP_IN);
       });
+    if (!zapInParams.from || !zapInParams.to || !zapInParams.data) throw new ZapError("zap in", ZapError.ZAP_IN);
 
-    const value = toBN(zapInParams.value).toFixed(0);
+    const value = toBN(zapInParams.value?.toString()).toFixed(0);
 
-    options.gasPrice = options.gasPrice || zapInParams.gasPrice;
+    options.gasPrice = options.gasPrice || zapInParams.gasPrice?.toString();
     if (!skipGasEstimate) {
-      options.gasLimit = zapInParams.gas;
+      options.gasLimit = zapInParams.gasLimit?.toString();
     }
 
     const tokensReceived = await this.simulationExecutor.simulateVaultInteraction(
       from,
       zapInParams.to,
-      zapInParams.data,
+      zapInParams.data as string,
```

Comment:
This casting looks suspicious, could we refactor to make sure it's either a `string` when it gets here or make the `simulateVaultInteraction` take that as whatever type it is and deal with it?

Reply:
Done with https://github.com/yearn/yearn-sdk/pull/301/commits/26bf68da796849e71dd2422916e584806b0010bb